openassessment/workflow/workflow_batch_update_api.py

```diff
@@ -0,0 +1,600 @@
+    """
+    Generates dictionary containing data required to update ORA workflows for all scopes.
+    This data structure is used as a local cache to avoid redundant DB queries during
+    batch workflow update process
+
+    Structure:
+    ```
+    {
+      "courses": [
+        {
+          "course_id": "value",
+          "course_settings": {},
+          "assessments": [
+            {
+              "item_id": "value",
+              "assessment_requirements": {},
+              "submissions": [
+                {
+                  "submission_uuid": "value"
+                }
+              ]
+            }
+          ]
+        }
+      ]
+    }
+    ```
+    """
+    workflow_update_data = {}
+    store = modulestore()
+    cache = set([])
+    for peer_workflow in peer_workflows:
+        try:
+            if peer_workflow.course_id not in cache:
+                # retrieve course block
+                course_block_key = UsageKey.from_string(peer_workflow.course_id)
+                course_block = store.get_item(course_block_key)
+
+                # retrieve openassessment block
+                ora_block_key = UsageKey.from_string(peer_workflow.item_id)
+                ora_block = store.get_item(ora_block_key)
+
+                if is_flexible_peer_grading_on(ora_block, course_block):
+                    # we are interested only in submissions for ORA with flexible peer grading configured
+                    course_settings = {
+                        'force_on_flexible_peer_openassessments': course_block.force_on_flexible_peer_openassessments}
+
+                    workflow_update_data = _add_course_data(
+                        workflow_update_data=workflow_update_data,
+                        course_id=peer_workflow.course_id,
+                        item_id=peer_workflow.item_id,
+                        submission_uuid=peer_workflow.submission_uuid,
+                        assessment_requirements=ora_block.workflow_requirements(),
+                        course_settings=course_settings)
+
+                cache.add(peer_workflow.course_id)
+                cache.add(peer_workflow.item_id)
+
+            elif peer_workflow.item_id not in cache:
+                # retrieve openassessment block
+                ora_block_key = UsageKey.from_string(peer_workflow.item_id)
+                ora_block = store.get_item(ora_block_key)
+
+                workflow_update_data = _add_ora_data(workflow_update_data=workflow_update_data,
+                                                     course_id=peer_workflow.course_id,
+                                                     item_id=peer_workflow.item_id,
+                                                     submission_uuid=peer_workflow.submission_uuid,
+                                                     assessment_requirements=ora_block.workflow_requirements())
+            else:
+                workflow_update_data = _add_submission_data(workflow_update_data,
```

Comment:
Same here, we just check if the item and course are in the cache, not if they are flexible.